scripts/generators/api_dump_generator.py

```diff
@@ -0,0 +1,1320 @@
+    if(fpCreateDevice == NULL) {
+        return VK_ERROR_INITIALIZATION_FAILED;
+    }
+
+    // Call the function and create the dispatch table
+    chain_info->u.pLayerInfo = chain_info->u.pLayerInfo->pNext;
+    VkResult result = fpCreateDevice(physicalDevice, pCreateInfo, pAllocator, pDevice);
+    if(result == VK_SUCCESS) {
+        initDeviceTable(*pDevice, fpGetDeviceProcAddr);
+    }
+
+    // Output the API dump
+    if (ApiDumpInstance::current().shouldDumpOutput()) {
+        switch(ApiDumpInstance::current().settings().format())
+        {
+            case ApiDumpFormat::Text:
+                dump_text_vkCreateDevice(ApiDumpInstance::current(), result, physicalDevice, pCreateInfo, pAllocator, pDevice);
+                break;
+            case ApiDumpFormat::Html:
+                dump_html_vkCreateDevice(ApiDumpInstance::current(), result, physicalDevice, pCreateInfo, pAllocator, pDevice);
+                break;
+            case ApiDumpFormat::Json:
+                dump_json_vkCreateDevice(ApiDumpInstance::current(), result, physicalDevice, pCreateInfo, pAllocator, pDevice);
+                break;
+        }
+    }
+    return result;
+}
+
+EXPORT_FUNCTION VKAPI_ATTR VkResult VKAPI_CALL vkEnumerateInstanceExtensionProperties(const char* pLayerName, uint32_t* pPropertyCount, VkExtensionProperties* pProperties)
+{
+    return util_GetExtensionProperties(0, NULL, pPropertyCount, pProperties);
+}
+
+EXPORT_FUNCTION VKAPI_ATTR VkResult VKAPI_CALL vkEnumerateInstanceLayerProperties(uint32_t* pPropertyCount, VkLayerProperties* pProperties)
+{
+    static const VkLayerProperties layerProperties[] = {
+        {
+            "VK_LAYER_LUNARG_api_dump",
+            VK_MAKE_VERSION(1, 4, VK_HEADER_VERSION), // specVersion
+            VK_MAKE_VERSION(0, 2, 0), // implementationVersion
+            "layer: api_dump",
+        }
+    };
+
+    return util_GetLayerProperties(ARRAY_SIZE(layerProperties), layerProperties, pPropertyCount, pProperties);
+}
+
+EXPORT_FUNCTION VKAPI_ATTR VkResult VKAPI_CALL vkEnumerateDeviceLayerProperties(VkPhysicalDevice physicalDevice, uint32_t* pPropertyCount, VkLayerProperties* pProperties)
+{
+    static const VkLayerProperties layerProperties[] = {
+        {
+            "VK_LAYER_LUNARG_api_dump",
+            VK_MAKE_VERSION(1, 4, VK_HEADER_VERSION),
+            VK_MAKE_VERSION(0, 2, 0),
+            "layer: api_dump",
+        }
+    };
+
+    return util_GetLayerProperties(ARRAY_SIZE(layerProperties), layerProperties, pPropertyCount, pProperties);
+}
+
+// Autogen instance functions
+''')
+
+        protect = PlatformGuardHelper()
+        for command in  [x for x in self.vk.commands.values() if x.instance]:
+            if command.name in ['vkCreateInstance', 'vkCreateDevice', 'vkGetInstanceProcAddr', 'vkEnumerateDeviceExtensionProperties', 'vkEnumerateDeviceLayerProperties', 'vkEnumerateInstanceExtensionProperties', 'vkEnumerateInstanceLayerProperties', 'vkEnumerateInstanceVersion']:
```

Comment:
this list feels random, it has some of the functions above, but also `vkEnumerateInstanceVersion`

maybe just add a comment with idea what is in here

Reply:
By moving the handwritten code into a separate file, I have a nice 'name' for the functions, and created a list to hold those function names.